pemFioi/conceptViewer-1.0-mobileFirst.js

```diff
@@ -470,6 +466,43 @@ function getConceptsFromBlocks(includeBlocks, allConcepts, context) {
   return concepts;
 }
 
+/**
+ * This function allow us to remove useless concept from the concept list.
+ * The "useless concepts" are concept who don't have a "python" list of all
+ * their functions or in which this list is empty.
+ * @param allConcepts All of the concepts.
+ * @return A new list of concepts but cleaned.
+ */
+function cleanAllConcept(allConcepts) {
```

Comment:
I have a few issues with that function `cleanAllConcept` :

* It seems pretty unnecessary to have a separate function. It is called only once, for one specific reason (filter out concepts which don't have a python attribute), and has low reusability, as I cannot really think of another scenario where we would want to filter out concepts without a python attribute. The whole function can be replaced by just one line in the function `findConceptByFunction`, which is the only function currently using it.
* The name is misleading : for instance, "filter" would be a better word than "clean", as doesn't really clean anything but just filters out some concepts from the list.
* The description is a bit misleading too : the concepts aren't "useless", they might be useless for a specific purpose, but not in general. You should explain why you call them useless, for instance "concepts which are useless for the function getConceptsFromBlocks"

I think you should just delete this function, and replace it with one line in findConceptByFunction.

Reply:
For the first point:
I usually do a lots of functions because I find it cleaner because the final code looks like it is written in english.

But I agree that this function is useless since it can be easily replaced by a simple javascript "filter" which takes only one lign.

I also changed the name of the variable to "filteredConcepts" instead of "cleanedConcepts".